common/src/main/java/org/opensearch/sdk/SearchDataObjectRequest.java

```diff
@@ -0,0 +1,88 @@
+/*
+ * Copyright OpenSearch Contributors
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * The OpenSearch Contributors require contributions made to
+ * this file be licensed under the Apache-2.0 license or a
+ * compatible open source license.
+ */
+package org.opensearch.sdk;
+
+import org.opensearch.search.builder.SearchSourceBuilder;
+
+public class SearchDataObjectRequest {
+
+    private final String[] indices;
+    private final SearchSourceBuilder searchSourceBuilder;
+
+    /**
+     * Instantiate this request with an optional list of indices and search source
+     * <p>
+     * For data storage implementations other than OpenSearch, an index may be referred to as a table
+     *
+     * @param indices             the indices to search for the object
```

Comment:
How this will work? After searching in multiple indices are we going to combine the results and then send it back through `SearchDataObjectResponse`?

Reply:
In practice in ML commons, there's a single index (the system index for connectors, models, etc.).

In general it can be an empty list (search all indices) or multiple indices.   The search request really just packages up the entire JSON of the search query and the full response as well, parsing it back out from the XContent.

So the search response will be identical (some metadata/stats and then a list of hits).